### PR TITLE
TASK-49420 : Mobile UI in perkstore

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: perk-store-webapps/src/main/webapp/css/main.less [hunk 1]
```diff
@@ -996,15 +1005,162 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
     .carousselParent, .carousselImage, .v-carousel__item, .v-window__container {
        max-height: 250px !important;
     }
+
+    .product-list {
+      .toolbarProduct {
+        padding: 15px 5px 15px 15px;
```
Comment (Rdidmohamedelhadi) [inline perk-store-webapps/src/main/webapp/css/main.less]: You can use pa-4 directly (https://vuetifyjs.com/en/styles/spacing/#how-it-works)

File: perk-store-webapps/src/main/webapp/css/main.less [hunk 2]
```diff
@@ -1014,15 +1170,82 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
       max-width: 50%;
       flex-basis: 50%;
     }
+    .product-list {
+      .toolbarProduct {
+        padding: 15px 5px 15px 15px;
```
Comment (Rdidmohamedelhadi) [inline perk-store-webapps/src/main/webapp/css/main.less]: the same remark for this line

File: perk-store-webapps/src/main/webapp/css/main.less [hunk 3]
```diff
@@ -996,15 +1005,162 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+          left: 0 ~'!important; /** orientation=rt */ ';
+          margin: 0 3%;
+          top: 10px;
+        }
+      }
+    }
   }
 }
-@media (max-width : @maxMobileWidth){
+@media (max-width : @smallMobileWidth){
```
Comment (Rdidmohamedelhadi) [inline perk-store-webapps/src/main/webapp/css/main.less]: it should not be changed

**Review by Rdidmohamedelhadi — CHANGES_REQUESTED** [review]
Preferably we use the predefined classes of vuetify for padding and margin

link : https://vuetifyjs.com/en/styles/spacing/#how-it-works

File: perk-store-webapps/src/main/webapp/css/main.less [hunk 4]
```diff
@@ -996,8 +1004,52 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+      .toolbarProduct {
+        .addProduct {
+          width: fit-content;
+        }
+        .filter {
+          width: 100% ;
+          margin: 0 -2%;
+          .perkStoreFilterField{
+            width: 60%;
+            margin-top: 15px;
```
Comment (Rdidmohamedelhadi) [inline perk-store-webapps/src/main/webapp/css/main.less:1018]: change this please to mt-4

File: perk-store-webapps/src/main/webapp/css/main.less [hunk 5]
```diff
@@ -1007,22 +1059,161 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+      .toolbarProduct {
+        .addProduct {
+          min-width: min-content;
+          margin: 0 -2%;
+        }
+        .filter {
+          width: 100% ;
+            .perkStoreFilterField{
+            width: 50%;
+            margin-top: 15px;
```
Comment (Rdidmohamedelhadi) [inline perk-store-webapps/src/main/webapp/css/main.less:1074]: margin-top: 15px ==> mt-4

File: perk-store-webapps/src/main/webapp/css/main.less [hunk 6]
```diff
@@ -1007,22 +1059,161 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+    .product-list {
+      .toolbarProduct {
+        .addProduct {
+          width: 45%
+        }
+        .filter {
+          width: 47% ;
+          .perkStoreFilterField{
+            width: 65%;
+            margin-top: 12px;
```
Comment (Rdidmohamedelhadi) [inline perk-store-webapps/src/main/webapp/css/main.less:1168]: margin-top: 12px; ==> mt-3

File: perk-store-webapps/src/main/webapp/css/main.less [hunk 7]
```diff
@@ -1033,6 +1224,30 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+        .spacer {
+          width: auto ;
+        }
+
+        .filter {
+          width: 25% ;
+          margin: 0 1% ;
+          .perkStoreFilterField{
+            width: 64%;
+            margin-top: 15px;
```
Comment (Rdidmohamedelhadi) [inline perk-store-webapps/src/main/webapp/css/main.less:1242]: same for this line

**Review by Rdidmohamedelhadi — CHANGES_REQUESTED** [review]

**Comment by GouadriaHanen** [comment]
@Rdidmohamedelhadi   they all depend on the media width!

**Comment by Rdidmohamedelhadi** [comment]
> @Rdidmohamedelhadi they all depend on the media width!

for this margin-top: 12px; you can use mt-3 (https://vuetifyjs.com/en/styles/spacing/#how-it-works)

**Comment by Rdidmohamedelhadi** [comment]
> @Rdidmohamedelhadi they all depend on the media width!

aaah ok !

**Review by Rdidmohamedelhadi — APPROVED** [review]
LGTM

**Pull request merged**